trace/trace.go

```diff
@@ -2,25 +2,61 @@ package trace
 
 import (
 	"context"
+	"crypto/rand"
+	"encoding/binary"
+	"encoding/hex"
+	mathrand "math/rand"
+	"net/http"
 	"sync"
 	"time"
 
-	"github.com/google/uuid"
 	"github.com/honeycombio/beeline-go/client"
-	"github.com/honeycombio/beeline-go/propagation"
 	"github.com/honeycombio/beeline-go/sample"
 	libhoney "github.com/honeycombio/libhoney-go"
+	"github.com/imdario/mergo"
 )
 
+// GlobalConfig stores configuration used throughout the beeline
 var GlobalConfig Config
 
+// HeaderSupplier is an interface that specifies methods for storing and retrieving
+// key value pairs. It is implemented by http.Headers.
+type HeaderSupplier interface {
+	Set(key string, value string)
+	Get(key string) string
+}
+
+// HTTPPropagator wraps the interface with the same name from the OpenTelemetry Go SDK.
+// It intentionally uses different method names to highlight the fact that it's API is
+// different than the analagous object in the OpenTelemetry Go SDK.
```

Comment:
If it's not the same interface, it feels misleading to give it the same name. I'm not sure what name to use; some brainstorming will come up with something, I'm sure.